### PR TITLE
[TR] PIM-5009: Add character counter in comment modal

### PR DESCRIPTION
**Pull request by willy-ahva**
| Q                 | A
| ----------------- | ---
| Specs             | no
| Behats            | no
| Blue CI           | Blue
| Changelog updated | no
| Review and 2 GTM  |
| Micro Demo (PO)   |
| Migration script  |
| Tech Doc          |

### CODE REVIEW

File: src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less [hunk 1]
```diff
@@ -222,8 +222,27 @@
...
+    .char-counter {
+      align-self: flex-end;
+    }
+
+    .char-counter.above {
+      color: red;
+    }
+
+    .char-counter.under {
+      color: black;
```
Comment (phaseinducer) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: Maybe you should use variables here. Ask @yulldesign.

File: src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less [hunk 2]
```diff
@@ -222,8 +222,27 @@
...
+    }
+  }
+}
+
+.modal-footer {
+  a.btn.ok.disabled {
+    pointer-events: none;
```
Comment (phaseinducer) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: What is it for ?

Comment (willy-ahva) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: disable the link. With it you can't click on the button to send your proposal

File: src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less [hunk 3]
```diff
@@ -222,8 +222,27 @@
...
+    .chars-counter {
+      align-self: flex-end;
+    }
+
+    .chars-counter.above {
+      color: red;
+    }
+
+    .chars-counter.under {
+      color: black;
```
Comment (grena) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: please use hexa

File: src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less [hunk 4]
```diff
@@ -222,8 +222,27 @@
...
+      margin: 10px 0;
       width: 100%;
     }
+
+    .chars-counter {
+      align-self: flex-end;
+    }
+
+    .chars-counter.above {
+      color: red;
```
Comment (grena) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: please use hexa

File: src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less [hunk 5]
```diff
@@ -222,8 +222,27 @@
...
+    .chars-counter {
+      align-self: flex-end;
+    }
+
+    .chars-counter.above {
+      color: #FF0000;
+    }
+
+    .chars-counter.under {
+      color: #000000;
```
Comment (phaseinducer) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: When I said "variables" I meant stuff like `@akeneo-grey`
See : https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/UIBundle/Resources/public/css/less/variables.less

Comment (grena) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: yep, use a Less variable for `.above` too please (must be an "error color") or something like that

**Comment by phaseinducer** [comment]
`pointer-events` css property is designed for svg elements, using it on other elements is experimental.
Anyway we should always deal with event-related stuff in js, not css. Let's save us some future headache. =)

File: src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less (hunk 5 above)
Comment (yulldesign) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: .above color is "@chesnut-red"
.under color is "@slate-grey"

**Comment by willy-ahva** [comment]
@phaseinducer done

**Comment by phaseinducer** [comment]
GTM !

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/form-modal.js [hunk 6]
```diff
@@ -117,8 +119,17 @@ define(
 
                         form.setElement(this.modal.$('.modal-body')).render();
 
+                        mediator.on('pim_enrich:form:modal:disable-ok-btn', function () {
+                            this.disableOkBtn();
+                        }.bind(this));
+                        mediator.on('pim_enrich:form:modal:enable-ok-btn', function () {
+                            this.enableOkBtn();
+                        }.bind(this));
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/form-modal.js]: please add spaces between mediator calls :)

**Comment by ClementGautier** [comment]
GTM

File: src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less [hunk 7]
```diff
@@ -222,8 +222,20 @@
...
-      margin: 10px 0 0;
+      margin: 10px 0;
       width: 100%;
     }
+
+    .chars-counter {
+      align-self: flex-end;
+    }
+
+    .chars-counter.above {
```
Comment (grena) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: `.disabled` ?

File: src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less [hunk 8]
```diff
@@ -222,8 +222,20 @@
...
+      color: @warningText;
+    }
+
+    .chars-counter.under {
```
Comment (grena) [inline src/Pim/Bundle/UIBundle/Resources/public/css/notifications.less]: `.disabled`?

**Comment by grena** [comment]
Ok, seen, some CSS stuff to put in EE, but GTM once done! nice

**Pull request merged**